### PR TITLE
Update the Keycloak quickstart to the new security support

### PR DESCRIPTION
**Pull request by cescoffier**

### CODE REVIEW

File: using-keycloak/src/main/java/org/acme/keycloak/AdminResource.java [hunk 1]
```diff
@@ -25,6 +27,7 @@
  * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
  */
 @Path("/api/admin")
+@Authenticated
```
Comment (stuartwdouglas) [inline using-keycloak/src/main/java/org/acme/keycloak/AdminResource.java]: This has not actually been wired up yet, it should be done before the next release though

Comment (cescoffier) [reply using-keycloak/src/main/java/org/acme/keycloak/AdminResource.java]: The question is: should it be there, or `@RoleAllowed` should be enough.

I believe `@RoleAllowed` should be enough. WDYT?

File: using-keycloak/src/main/java/org/acme/keycloak/UsersResource.java [hunk 2]
```diff
@@ -15,42 +15,40 @@
...
     }
 
     public class User {
 
         private final String userName;
 
-        User(KeycloakSecurityContext securityContext) {
-            this.userName = securityContext.getToken().getPreferredUsername();
+        User(SecurityIdentity identity) {
+            this.userName = identity.getPrincipal().getName();
```
Comment (sberyozkin) [inline using-keycloak/src/main/java/org/acme/keycloak/UsersResource.java]: Is the access to the token has been lost ?

Comment (stuartwdouglas) [reply using-keycloak/src/main/java/org/acme/keycloak/UsersResource.java]: In this case the Principal is an instance of JsonWebToken

File: using-keycloak/src/main/java/org/acme/keycloak/AdminResource.java (hunk 1 above)
Comment (stuartwdouglas) [reply using-keycloak/src/main/java/org/acme/keycloak/AdminResource.java]: @RolesAllowed("*") is the same

**Pull request merged**